src/contracts/core/DelegationManager.sol

```diff
@@ -226,15 +226,18 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
             }
         }
 
-        // actually undelegate the staker
-        _undelegate(staker);
+        // actually undelegate the staker -- only make storage changes + emit an event if the staker is actively delegated, otherwise do nothing
+        if (isDelegated(staker)) {
+            emit StakerUndelegated(staker, operator);
```

Comment:
Do we need `if (isDelegated(staker))`? Surely that's enforced above, and we would want this to fail if false.

Hmm, looking above, I think there may not be a hard requirement? Let's add `require(operator != 0)` or `require(isDelegated(staker))`

Reply:
yeah, there's documentation on this being a no-op in this case, but I agree given the other changes that this require statement makes sense to add now.

Reply:
addressed this in a commit. https://github.com/Layr-Labs/eigenlayer-contracts/pull/183/commits/be217b232c368f700dc4cb7cbacf5a805cdd450d